inject-java/src/main/java/io/micronaut/annotation/processing/visitor/JavaMethodElement.java

```diff
@@ -270,4 +274,29 @@ private boolean isSuspend(VariableElement ve) {
         return false;
     }
 
+    private Element getFieldElementForWriter() {
+        String[] writerPrefixes = getAnnotationMetadata()
+                .getValue(AccessorsStyle.class, "writePrefixes", String[].class)
+                .orElse(new String[]{AccessorsStyle.DEFAULT_WRITE_PREFIX});
+
+        final String methodName = getName();
+        if (!NameUtils.isWriterName(methodName, writerPrefixes) || executableElement.getParameters().size() != 1) {
+            return null;    // not a writer
+        }
+
+        Element classElement = executableElement.getEnclosingElement();
+        if (!(classElement instanceof TypeElement)) {
+            return null;    // not within a class
+        }
+
+        final String expectedFieldName = NameUtils.getPropertyNameForSetter(methodName, writerPrefixes);
+
+        // Return the field corresponding to this writer.
+        return classElement.getEnclosedElements()
+                .stream()
+                .filter(element -> element.getKind() == ElementKind.FIELD)
+                .filter(field -> field.getSimpleName().contentEquals(expectedFieldName))
+                .findFirst()
+                .orElse(null);
```

Comment:
This search doesn't make sense, should be use the `PropertyElement` API and `getBeanProperties`

Reply:
Sorry, I don't follow. I don't see how to extract a field `javax.lang.model.element.Element` from the `PropertyElement`/`getBeanProperties` API.